### DISCUSSION TITLE
How can I fix an extraction error with Swiftgram's TDLib framework during build?

### DISCUSSION THREAD

**Discussion by S0ulf3re** [mock-foundation/moc#37] [Q&A] [answered]
I really want to try and help contribute this project because I personally don't like the default telegram client for Mac and I've been trying to learn swift ever since July of last year. I tried opening the project, but I got an error which has caused a bunch of other build time errors to occur. How can I fix all this?

Here's a log from Xcode
[Resolve Packages_2022-01-26T12-44-01.txt](https://github.com/ggoraa/moc/files/7944050/Resolve.Packages_2022-01-26T12-44-01.txt)

**Comment by voltangle** [comment]
The issue should not exist, it builds on my machine and on CI, can you check whether `xcodebuild -scheme Moc clean build` works? You should run this command in the project directory in terminal

**Reply by S0ulf3re** [reply]
I managed to fix this all by redownloading the project, and I managed to get the project compiled. I had to temporarily disable SwiftLint by commenting it out because I'm having issues with it giving nonzero results about it not being found. Despite it being installed with homebrew (I will fix it and turn it back on should I make any contributions)


Next thing I'm trying to do is set up the API hash and ID. But it doesn't seem to be working. I'm not sure if the app title has anything to do with it or if I just did something wrong.

**Reply by voltangle** [reply]
What errors are you getting in the console(if you ran the app)? 
And yes, the app title in https://my.telegram.org is just for the looks, how your client will be displayed in devices list

About SwiftLint and Homebrew - did you try invoking `swiftlint` from the CLI? If there is an error like "command not found", then there is some issues with the installation

**Reply by voltangle** [reply]
I found the cause
It was bad configuration of build phases
I pushed an update to master branch, it should fix the issue

**Reply by S0ulf3re** [reply]
Just a quick question, Are you using homebrew for things or are you compiling everything from source? because I just downloaded the new project and went and installed gyb with `brew install nshipster/formulae/gyb` and I can run it from the command line, but Xcode can't seem to find it and it just keeps throwing back a nonzero exit code.

**Reply by voltangle** [reply]
I am usually using Homebrew for installing these kinds of things, but when it is not available, I manually install them
About a nonzero exit code, well, I actually have no idea why it fails, it should run it
And yes, can you give me more info about that nonzero exit code? Maybe I can find the cause by looking at logs

**Reply by S0ulf3re** [reply]
It was complaining about commands being not found. I managed to fix it by specifying SwiftLint and gyb to the full paths (which for me is /opt/homebrew/bin/)

**Reply by voltangle** [reply]
hm, that's strange
anyways, it works, so problem resolved 😁

**Comment by S0ulf3re** [comment, marked as answer]
Ok, so I just cleared both caches within library and Xcode, and restarted my computer. Then when I opened it, Xcode started complaining about TDlib framework. So I cleared the package cache. Then, it started complaining about SystemUtils and ImageUtils, and I managed to fix that by restarting Xcode.

In the end, it is now once again complaining about commands not being found, specifically gyb
[Build target Moc_2022-01-27T10-16-39.txt](https://github.com/ggoraa/moc/files/7951527/Build.target.Moc_2022-01-27T10-16-39.txt)

**Reply by voltangle** [reply]
Anyways, you can add these absolute paths to your local copy, but do not push that to the remote
And when you do that, you will be ready to contribute to Moc! :D
